scripts/compile.sh

```diff
@@ -0,0 +1,43 @@
+# compile.sh looks through a directory and compiles all of the tests
```

Comment:
Should we using go instead of bash here? @SaxyPandaBear  @adam-mateen

Reply:
I think there's enough logic in here that would remain bash, even if I rewrote this in go. So I would be reluctant to switch, but open to suggestion

Reply:
We do not need a bash script to compile test files. 
```bash
go test -run=nope -v ./...
```
This looks at all of the folders recursively, and attempts to execute any test that matches `nope`, which is hopefully none of them, but still compiles all of the test files.

Reply:
Huh that's a really good hack

```
go test -run=nope -v ./...     
# github.com/aws/amazon-cloudwatch-agent-test/test/metric_value_benchmark
test/metric_value_benchmark/disk_test.go:39:1: expected declaration, found asdfasfd
FAIL    github.com/aws/amazon-cloudwatch-agent-test/test/metric_value_benchmark [setup failed]
?       github.com/aws/amazon-cloudwatch-agent-test/environment [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/environment/computetype     [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/environment/ecsdeploymenttype       [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/environment/ecslaunchtype   [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/filesystem  [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/generator   [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/install     [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/internal/common     [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/msi/tools/msiversion        [no test files]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/ca_bundle      (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/canary (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/cloudwatchlogs (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/collection_interval    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/ecs/ecs_metadata       (cached) [no tests to run]
?       github.com/aws/amazon-cloudwatch-agent-test/test/metric [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension       [no test files]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/metric_dimension       (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/metrics_number_dimension       (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/performancetest        (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/run_as_user    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/aws/amazon-cloudwatch-agent-test/test/sanity (cached) [no tests to run]
?       github.com/aws/amazon-cloudwatch-agent-test/test/status [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/test/test_runner    [no test files]
FAIL
```

I think I'd prefer `go test -run=nope ./test/...` to declutter the logs

```
go test -run=nope ./test/...
# github.com/aws/amazon-cloudwatch-agent-test/test/metric_value_benchmark
test/metric_value_benchmark/disk_test.go:39:1: expected declaration, found asdfasfd
FAIL    github.com/aws/amazon-cloudwatch-agent-test/test/metric_value_benchmark [setup failed]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/ca_bundle      (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/canary (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/cloudwatchlogs (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/collection_interval    (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/ecs/ecs_metadata       (cached) [no tests to run]
?       github.com/aws/amazon-cloudwatch-agent-test/test/metric [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension       [no test files]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/metric_dimension       (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/metrics_number_dimension       (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/performancetest        (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/run_as_user    (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/sanity (cached) [no tests to run]
?       github.com/aws/amazon-cloudwatch-agent-test/test/status [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/test/test_runner    [no test files]
FAIL

```

example with an error

```
go test -run=nope ./test/...
# github.com/aws/amazon-cloudwatch-agent-test/test/metric_value_benchmark
test/metric_value_benchmark/disk_test.go:39:1: expected declaration, found asdfasfd
FAIL    github.com/aws/amazon-cloudwatch-agent-test/test/metric_value_benchmark [setup failed]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/ca_bundle      (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/canary (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/cloudwatchlogs (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/collection_interval    (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/ecs/ecs_metadata       (cached) [no tests to run]
?       github.com/aws/amazon-cloudwatch-agent-test/test/metric [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension       [no test files]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/metric_dimension       (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/metrics_number_dimension       (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/performancetest        (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/run_as_user    (cached) [no tests to run]
ok      github.com/aws/amazon-cloudwatch-agent-test/test/sanity (cached) [no tests to run]
?       github.com/aws/amazon-cloudwatch-agent-test/test/status [no test files]
?       github.com/aws/amazon-cloudwatch-agent-test/test/test_runner    [no test files]
FAIL

```